### ISSUE TITLE
Build failure for 0.8.8.1 on ghc 9.2.1

### ISSUE THREAD

**Issue by jgm** [haskell-compat/base-orphans#66]
This just came up in CI.
See https://github.com/jgm/texmath/actions/runs/4350285790/jobs/7600847303#step:6:222
e.g.
```
src/Data/Orphans.hs:1859:19: error:
    Not in scope: type constructor or class ‘Ix’
    Perhaps you meant ‘IO’ (imported from Prelude)
```

**Comment by RyanGlScott** [comment]
Ugh, I didn't notice this at the time because `base-orphans-0.8.8.1` _does_ build successfully with GHC 9.2.7. The difference is that GHC 9.2.1 ships with a different version of `base` that exposes fewer instances, and the CPP in `base-orphans` is sensitive to this.

I think I see how to fix this—I'll put up a patch later this afternoon.

**Comment by jgm** [comment]
And I'll change my CI to use 9.2.7.  But I guess it's good that my old version flushed out this issue.

**Comment by RyanGlScott** [comment]
I've fixed this in 25cb66b3a4682f47660b25e7db9d56401c8a06db and uploaded the changes to Hackage as [`base-orphans-0.8.8.2`](https://hackage.haskell.org/package/base-orphans-0.8.8.2).
